src/sqs/index.ts

```diff
@@ -40,24 +51,60 @@ export type QueueSettings = {
 
 	readonly longPollingIntervalSec?: number,
 
-	//TODO Add batching
+	/**
+	 * Timeout for processing a single message in seconds
+	 */
+	readonly timeoutSec: number;
 
-	//TODO Add error handling
+	/**
+	 * Defines how many times the message can be attempted to be executed
+	 */
+	readonly maxAttempts: number;
+
+	//TODO Add batching
 
-	//TODO Add message processing timeouts
 }
 
 const DEFAULT_LONG_POLLING_INTERVAL = 4
 
 const PROCESSING_DURATION_HISTOGRAM_BUCKETS =	"10_100_500_1000_2000_3000_5000_10000_30000_60000";
 
+/**
+ * Error indicating a timeout
+ */
+export class SqsTimeoutError extends Error {
+}
 /**
  * Handler for the queue messages
  */
 export interface MessageHandler<MessagePayload> {
 	handle(context: Context<MessagePayload>): Promise<void>;
 }
 
+export interface ErrorHandler<MessagePayload> {
+	handle(error: Error, context: Context<MessagePayload>): ErrorHandlingResult;
```

Comment:
we know for sure that the error handler will be synchronous?

Reply:
Good question.

In case if we want to use a feature flag inside an error handler we'll have to make it async.

Let's do it async.